src/message.js

```diff
@@ -351,23 +352,24 @@ export class Message {
 
   /**
    * Encrypt a session key either with public keys, passwords, or both at once.
-   * @param  {Uint8Array} sessionKey     session key for encryption
-   * @param  {String} algorithm          session key algorithm
-   * @param  {String} aeadAlgorithm      (optional) aead algorithm, e.g. 'eax' or 'ocb'
-   * @param  {Array<Key>} publicKeys     (optional) public key(s) for message encryption
-   * @param  {Array<String>} passwords   (optional) for message encryption
-   * @param  {Boolean} wildcard          (optional) use a key ID of 0 instead of the public key IDs
-   * @param  {Date} date                 (optional) override the date
-   * @param  {Array} userIds             (optional) user IDs to encrypt for, e.g. [{ name:'Robert Receiver', email:'robert@openpgp.org' }]
-   * @returns {Promise<Message>}          new message with encrypted content
+   * @param  {Uint8Array} sessionKey                      session key for encryption
+   * @param  {String} algorithm                           session key algorithm
+   * @param  {String} aeadAlgorithm                       (optional) aead algorithm, e.g. 'eax' or 'ocb'
+   * @param  {Array<Key>} publicKeys                      (optional) public key(s) for message encryption
+   * @param  {Array<String>} passwords                    (optional) for message encryption
+   * @param  {Boolean} wildcard                           (optional) use a key ID of 0 instead of the public key IDs
+   * @param  {Date} date                                  (optional) override the date
+   * @param  {Array} userIds                              (optional) user IDs to encrypt for, e.g. [{ name:'Robert Receiver', email:'robert@openpgp.org' }]
+   * @param  {Array<module:type/keyid>} encryptionKeyIds  (optional) array of key IDs to use for encryption. Each encryptionKeyIds[i] corresponds to publicKeys[i]
+   * @returns {Promise<Message>}                          new message with encrypted content
    * @async
    */
-  static async encryptSessionKey(sessionKey, algorithm, aeadAlgorithm, publicKeys, passwords, wildcard = false, date = new Date(), userIds = []) {
+  static async encryptSessionKey(sessionKey, algorithm, aeadAlgorithm, publicKeys, passwords, wildcard = false, date = new Date(), userIds = [], encryptionKeyIds = []) {
     const packetlist = new PacketList();
 
     if (publicKeys) {
-      const results = await Promise.all(publicKeys.map(async function(publicKey) {
-        const encryptionKey = await publicKey.getEncryptionKey(undefined, date, userIds);
+      const results = await Promise.all(publicKeys.map(async function(publicKey, index) {
+        const encryptionKey = await publicKey.getEncryptionKey(encryptionKeyIds[index], date, userIds);
```

Comment:
This is also very nitpicky but I would just call this variable `i`, for symmetry with `sign` below.
```suggestion
      const results = await Promise.all(publicKeys.map(async function(publicKey, i) {
        const encryptionKey = await publicKey.getEncryptionKey(encryptionKeyIds[i], date, userIds);
```